dune_client/api/extensions.py

```diff
@@ -0,0 +1,196 @@
+            params={
+                "table_name": table_name,
+                "description": description,
+                "data": data,
+            },
+        )
+        try:
+            return bool(response_json["success"])
+        except KeyError as err:
+            raise DuneError(response_json, "upload_csv response", err) from err
+
+    ######################
+    # Deprecated Functions
+    ######################
+    @deprecated(version="1.2.1", reason="Please use run_query")
+    def refresh(
+        self,
+        query: QueryBase,
+        ping_frequency: int = 5,
+        performance: Optional[str] = None,
+    ) -> ResultsResponse:
+        """
+        Executes a Dune `query`, waits until execution completes,
+        fetches and returns the results.
+        Sleeps `ping_frequency` seconds between each status request.
+        """
+        return self.run_query(query, ping_frequency, performance)
+
+    @deprecated(version="1.2.1", reason="Please use run_query_csv")
+    def refresh_csv(
+        self,
+        query: QueryBase,
+        ping_frequency: int = 5,
+        performance: Optional[str] = None,
+    ) -> ExecutionResultCSV:
+        """
+        Executes a Dune query, waits till execution completes,
+        fetches and the results in CSV format
+        (use it load the data directly in pandas.from_csv() or similar frameworks)
+        """
+        return self.run_query_csv(query, ping_frequency, performance)
+
+    @deprecated(version="1.2.1", reason="Please use run_query_dataframe")
+    def refresh_into_dataframe(
+        self, query: QueryBase, performance: Optional[str] = None
+    ) -> Any:
+        """
+        Execute a Dune Query, waits till execution completes,
+        fetched and returns the result as a Pandas DataFrame
+
+        This is a convenience method that uses refresh_csv underneath
+        """
+        return self.run_query_dataframe(query, performance)
+
+    #################
+    # Private Methods
+    #################
+    def _refresh(
+        self,
+        query: QueryBase,
+        ping_frequency: int = 5,
+        performance: Optional[str] = None,
+    ) -> str:
+        """
+        Executes a Dune `query`, waits until execution completes,
+        fetches and returns the results.
+        Sleeps `ping_frequency` seconds between each status request.
+        """
+        job_id = self.execute_query(query=query, performance=performance).execution_id
+        status = self.get_execution_status(job_id)
+        while status.state not in ExecutionState.terminal_states():
+            self.logger.info(
+                f"waiting for query execution {job_id} to complete: {status}"
+            )
+            time.sleep(ping_frequency)
+            status = self.get_execution_status(job_id)
+        if status.state == ExecutionState.FAILED:
+            self.logger.error(status)
+            raise QueryFailed(f"{status}. Perhaps your query took too long to run!")
```

Comment:
This is just a legacy message since the only reason I have every reached this point was because of query timeouts (30 minutes). However there are going to be many reasons why queries will fail especially if people start haphazardly using the `run_sql` feature with bad code. I would have to test and see what kinds of responses we get back for invalid queries (maybe its the same types of errors we see in the interface). In any case, it would be a good issue to make this part better.

Reply:
I I made myself a [failing query](https://dune.com/queries/3008125) and checked the response `JOB_ID=01H9STYJ63ETQKCQ8PVG3NMREM`. 

```
{"execution_id":"01H9STYJ63ETQKCQ8PVG3NMREM","query_id":3008125,"state":"QUERY_STATE_FAILED","submitted_at":"2023-09-08T07:26:21.893177Z","expires_at":"2023-12-07T07:26:22.095658Z","execution_started_at":"2023-09-08T07:26:21.907079Z","execution_ended_at":"2023-09-08T07:26:22.095657Z","error":{"type":"FAILED_TYPE_EXECUTION_FAILED","message":"line 1:1: mismatched input 'selcjthehg'. Expecting: 'ALTER', 'ANALYZE', 'CALL', 'COMMENT', 'COMMIT', 'CREATE', 'DEALLOCATE', 'DELETE', 'DENY', 'DESC', 'DESCRIBE', 'DROP', 'EXECUTE', 'EXPLAIN', 'GRANT', 'INSERT', 'MERGE', 'PREPARE', 'REFRESH', 'RESET', 'REVOKE', 'ROLLBACK', 'SET', 'SHOW', 'START', 'TRUNCATE', 'UPDATE', 'USE', \u003cquery\u003e at line 1, position 1 [Execution ID: 01H9STYJ63ETQKCQ8PVG3NMREM]","metadata":{"line":1,"column":1}}}
```

Looks like we could build the `QueryFailed` error type out of `ExecutionResponse["error"]`. I will create an issue out of this.